pytensor/tensor/basic.py

```diff
@@ -3726,109 +3727,19 @@ def trace(a, offset=0, axis1=0, axis2=1):
     return diagonal(a, offset=offset, axis1=axis1, axis2=axis2).sum(-1)
 
 
-class AllocDiag(Op):
-    """An `Op` that copies a vector to the diagonal of a zero-ed matrix."""
+class AllocDiag(OpFromGraph):
```

Comment:
I would perhaps still want to inline it but during specialization when the linalg rewrites are not looking for it anymore? The logic you had to register an inline rewrite later still makes sense in that case

Reply:
You mean just keeping inline as a prop? Or re-implementing the JAX now everything) rewrite that eliminates the OpFromGraph completely?

I'm not sure clear on what the computational benefits of inline vs not are

Reply:
No, because we don't want to inline in jax. We need a custom rewrite to inline this Op that's excluded in JAX mode

Reply:
Sorry about the last message here, may have confused with pad. For AllocDiag we can always inline late, even in the JAX backend. It was for pad that jax was complaining and we needed to dispatch on their impl directly